src/authorization/AcpUtil.ts

```diff
@@ -0,0 +1,101 @@
+import type { IAccessControl } from '@solid/access-control-policy/dist/type/i_access_control';
+import type { IAccessControlResource } from '@solid/access-control-policy/dist/type/i_access_control_resource';
+import type { IAccessControlledResource } from '@solid/access-control-policy/dist/type/i_access_controlled_resource';
+import type { IAccessMode } from '@solid/access-control-policy/dist/type/i_access_mode';
+import type { IMatcher } from '@solid/access-control-policy/dist/type/i_matcher';
+import type { IPolicy } from '@solid/access-control-policy/dist/type/i_policy';
+import type { Store } from 'n3';
+import type { NamedNode, Term } from 'rdf-js';
+import { ACP } from '../util/Vocabularies';
+
+/**
+ * Returns all objects found using the given subject and predicate, mapped with the given function.
+ */
+function mapObjects<T>(data: Store, subject: Term, predicate: Term, fn: (data: Store, term: Term) => T): T[] {
+  return data.getObjects(subject, predicate, null)
+    .map((term): T => fn(data, term));
+}
+
+/**
+ * Returns the string values of all objects found using the given subject and predicate.
+ */
+function getObjectValues(data: Store, subject: Term, predicate: NamedNode): string[] {
+  return mapObjects(data, subject, predicate, (unused, term): string => term.value);
+}
+
+/**
+ * Finds the {@link IMatcher} with the given identifier in the given dataset.
+ * @param data - Dataset to look in.
+ * @param matcher - Identifier of the matcher.
+ */
+export function getMatcher(data: Store, matcher: Term): IMatcher {
```

Comment:
Mostly just because I felt the `AcpReader` class was getting bloated with all this in it. Now the parsing of the document is separated from the interpretation.

Reply:
Mmm but it's not really a reusable utility. Separate parser then? Up to you.